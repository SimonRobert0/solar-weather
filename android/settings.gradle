rootProject.name = 'solar'
include ':react-native-google-places'
project(':react-native-google-places').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-google-places/android')
include ':react-native-config'
project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
include ':realm'
project(':realm').projectDir = new File(rootProject.projectDir, '../node_modules/realm/android')

include ':app'
