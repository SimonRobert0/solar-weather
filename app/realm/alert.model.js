import Realm from 'realm';

class Alert extends Realm.Object {}
Alert.schema = {
  name: 'Alert',
  properties: {
    title: 'string',
    time: 'int',
    severity: 'string',
    expires: 'int',
    description: 'string',
    uri: 'string',
  },
};

export default Alert;
