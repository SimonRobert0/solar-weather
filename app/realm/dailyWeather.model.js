import Realm from 'realm';

class DailyData extends Realm.Object {}
DailyData.schema = {
  name: 'DailyData',
  properties: {
    time: 'int',
    summary: 'string',
    icon: 'string',
    sunriseTime: 'int',
    sunsetTime: 'int',
    precipIntensity: 'float',
    precipProbability: 'float',
    precipType: { type: 'string', optional: true },
    temperatureMin: 'float',
    temperatureMax: 'float',
    humidity: 'float',
  },
};

class Daily extends Realm.Object {}
Daily.schema = {
  name: 'Daily',
  properties: {
    summary: 'string',
    icon: 'string',
    data: { type: 'list', objectType: 'DailyData' },
  },
};

module.exports = {
  Daily,
  DailyData,
};
