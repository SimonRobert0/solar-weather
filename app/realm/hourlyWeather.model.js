import Realm from 'realm';

class HourlyData extends Realm.Object {}
HourlyData.schema = {
  name: 'HourlyData',
  properties: {
    time: 'int',
    summary: 'string',
    icon: 'string',
    precipIntensity: 'float',
    precipProbability: 'float',
    precipType: { type: 'string', optional: true },
    temperature: 'float',
    humidity: 'float',
  },
};

class Hourly extends Realm.Object {}
Hourly.schema = {
  name: 'Hourly',
  properties: {
    summary: 'string',
    icon: 'string',
    data: { type: 'list', objectType: 'HourlyData' },
  },
};

module.exports = {
  Hourly,
  HourlyData,
};
