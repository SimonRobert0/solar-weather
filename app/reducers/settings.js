import moment from 'moment';
import * as types from '../../lib/js/app/actions/actions';

const Maybe = require('data.maybe');
const R = require('ramda');

export const initialState = {
	unit: 'c',
	unitIndex: 0,
	timeType: '24',
	timeIndex: 0,
	onboarding: false,
	locationIndex: 0,
	latestUpdate: moment()
		.subtract(1, 'day')
		.unix()
		.toString(),
};

const updateField = (state, key, value) => ({
	...state,
	[key]: value,
});

const actionHandlers = {};
actionHandlers[types.setOnboarding] = (state, action) => ({
	...updateField(state, 'onboarding', action.value),
});

actionHandlers[types.setUnit] = (state, action) => ({
	...state,
	unit: action.unit,
	unitIndex: action.unitIndex,
});

actionHandlers[types.setActiveLocation] = (state, action) => ({
	...updateField(state, 'locationIndex', action.index),
});

actionHandlers[types.updateLatestTimestamp] = (state, action) => ({
	...updateField(state, 'latestUpdate', action.timestamp),
});

actionHandlers[types.setTimeType] = (state, action) => ({
	...state,
	timeType: action.timeType,
	timeIndex: action.timeIndex,
});

actionHandlers[types.setSettings] = (state, action) => ({
	...state,
	...R.dissoc('type', action),
});

export default function settings(state = initialState, action = {}) {
	return Maybe.fromNullable(actionHandlers[action.type])
		.map(handler => handler(state, action))
		.getOrElse(state);
}
