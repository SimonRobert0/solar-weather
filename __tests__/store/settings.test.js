import mockStore from 'redux-mock-store';
import { setTimeType, setUnit } from '../../app/actions/settings.action';
import { deleteRealm } from '../../app/realm';

describe('settings store', () => {
	const store = mockStore();

	beforeAll(async () => {
		jest.genMockFromModule('realm');
		// Teardown Realm prior to running tests
		await deleteRealm();
	});

	afterAll(async () => {
		// Clean up after adding to Realm
		await deleteRealm();
	});

	beforeEach(() => {
		store.clearActions();
	});

	it('should handle setTimeType', async () => {
		await store.dispatch(setTimeType('24', 1));
		expect(store.getActions()).toMatchSnapshot();
	});

	it('should handle setUnit', async () => {
		await store.dispatch(setUnit('f', 1));
		expect(store.getActions()).toMatchSnapshot();
	});
});
