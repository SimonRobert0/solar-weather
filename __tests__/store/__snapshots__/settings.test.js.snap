// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings store should handle setTimeType 1`] = `
Array [
  Object {
    "timeIndex": 1,
    "timeType": "24",
    "type": "SET_TIME_TYPE",
  },
]
`;

exports[`settings store should handle setUnit 1`] = `
Array [
  Object {
    "type": "SET_UNIT",
    "unit": "f",
    "unitIndex": 1,
  },
]
`;
